### PR TITLE
github: publish user info for token hashes

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Currently, GitHub API interactions are sharded by the hash of the
authorization token used to make them, in order to track between
accounts that may be using ghproxy concurrently. This new metric allows
for a join on labels to identify those users by name.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @alvaroaleman @cjw

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
/assign @cjwagner 

Dove-tails with https://github.com/kubernetes/test-infra/pull/17157

File: prow/github/client.go [hunk 1]
```diff
@@ -811,6 +827,14 @@ func (c *client) getUserData() error {
 	// email needs to be publicly accessible via the profile
 	// of the current account. Read below for more info
 	// https://developer.github.com/v3/users/#get-a-single-user
+
+	// record information for the user
+	authHeader := bytes.NewBufferString("Token ")
+	authHeader.Write(c.getToken())
+	hasher := sha256.New()
+	hasher.Write(authHeader.Bytes())
```
Comment (alvaroaleman) [inline prow/github/client.go:835]: Why are we not simply using `crypto/sha256.Sum256([]byte(fmt.Sprintf("Token: %s@, c.getToken())))`?

Comment (stevekuznetsov) [reply prow/github/client.go:835]: Is this python?!

Comment (stevekuznetsov) [reply prow/github/client.go:835]: I copy-pasted this from how I did it in `ghmetrics.go` -- if the `Sum256` is identical, I can refactor these two to do that. WDYT?

Comment (alvaroaleman) [reply prow/github/client.go:835]: yeah please do

Comment (stevekuznetsov) [reply prow/github/client.go]: updated

**Pull request merged**